pkg/services/govmomi/vcenter/clone.go

```diff
@@ -107,11 +108,6 @@ func Clone(ctx *context.VMContext, bootstrapData []byte) error {
 		return errors.Wrapf(err, "unable to get folder for %q", ctx)
 	}
 
-	datastore, err := ctx.Session.Finder.DatastoreOrDefault(ctx, ctx.VSphereVM.Spec.Datastore)
-	if err != nil {
-		return errors.Wrapf(err, "unable to get datastore for %q", ctx)
-	}
-
```

Comment:
This would cause the datastore to not be set if there is no `StoragePolicyName` specified. Is that intentional?